src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala

```diff
@@ -350,8 +350,15 @@ private class AppTaskLauncherActor(
       }
 
       val taskId = taskOp.taskId
-      tasksMap += taskId -> taskOp.newTask
-      inFlightTaskOperations += taskId -> None
+      taskOp.maybeNewTask match {
+        case Some(newTask) =>
+          tasksMap += taskId -> newTask
+          // Why?
```

Comment:
I don't get "how many" but I also do not get the code below the "Why?" but I didn't change it.